Add markdown description for String.LastIndexOf() method (#1952)
Codecademy/docs: content/c-sharp/concepts/strings/terms/lastindexof/lastindexof.md

---
Title: '.LastIndexOf()'

> Description: 'Find the last occurrence of a specified character or substring within a string'

Reviewer: ```suggestion
Description: 'Finds the last occurrence of a specified character or substring within a string.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Characters'
  - 'Methods'
CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

The **`.LastIndexOf()`** method returns the last occurrence of a specified character or substring within a `string`.

> The method return `-1` if there's not such valid search results.

Reviewer: ```suggestion
The method returns `-1` if there's no such match.
```

> An additional parameters may be supplied to define the criteria for the search.

Reviewer: ```suggestion
Additional parameters may be supplied to define the criteria for the search.
```

String.LastIndexOf(char);

// Same as above but uses additional starting position. The search starts at the startIndex and proceeds backward toward the beginning of the string.
String.LastIndexOf(char, startIndex);

// Same as above but uses additional searching length. The search starts at the startIndex and proceeds backward toward the beginning of the string for a specified number of character positions.
String.LastIndexOf(char, startIndex, length);

// Determines the zero-base index of the last occurrence of substring within the string
String.LastIndexOf(substring);

// Same as above but uses additional starting position. The search starts at the startIndex and proceeds backward toward the beginning of the string.
String.LastIndexOf(substring, startIndex);

// Same as above but uses additional searching length. The search starts at the startIndex and proceeds backward toward the beginning of the string for a specified number of character positions.
String.LastIndexOf(string, startIndex, length);

// Determines the zero-base index of the last occurrence of substring within the string, using the stringComparison method.
String.LastIndexOf(substring, stringComparison);

> // Same as about but use additional startIndex. The search starts at the startIndex and proceeds backward toward the beginning of the string, using the stringComparison method.

Reviewer: ```suggestion
// Same as above but use additional startIndex. The search starts at the startIndex and proceeds backward toward the beginning of the string, using the stringComparison method.
```

String.LastIndexOf(substring, startIndex, stringComparison);

> // Same as about but use additional searching length. The search starts at the startIndex and proceeds backward toward the beginning of the string for a specified number of character positions, using the stringComparison method.

Reviewer: ```suggestion
// Same as above but uses additional searching length. The search starts at the startIndex and proceeds backward toward the beginning of the string for a specified number of character positions, using the stringComparison method.
```

> String.LastIndexOf(substring, startIndex, lenth, stringComparison);

Reviewer: ```suggestion
String.LastIndexOf(substring, startIndex, length, stringComparison);
```

```

`.LastIndexOf()` is a method of the `String` object. It may take following parameters:

- `char` is an instance of the Char structure; represents a single letter.
- `string` is an instance of the String object.
- `startIndex` is an Int32 object.

> - `len` is an Int32 object.

Reviewer: ```suggestion
- `length` is an Int32 object.
```

        result = baseString.LastIndexOf(compareStr);
        Console.WriteLine("Simple search: {0}", result.ToString());
        
        // String.LastIndexOf(string, startIndex)
        result = baseString.LastIndexOf(compareStr, startIndex);
        Console.WriteLine("Search using starting postion: {0}", result.ToString());
        
        // String.LastIndexOf(string, startIndex, length)
        result = baseString.LastIndexOf(compareStr, startIndex, searchLength);
        Console.WriteLine("Search using starting postion & searching length: {0}", result.ToString());
        
        // String.LastIndexOf(string, startIndex, length, stringComparison)
        compareStr = "C";
        StringComparison strCmp = StringComparison.InvariantCultureIgnoreCase;
        result = baseString.LastIndexOf(compareStr, startIndex, strCmp);
        Console.WriteLine("Search using starting postion & string comparision: {0}", result.ToString());
    }
}
```

This is what is printed to the console:

```shell
Simple search: 19
Search using starting postion: 9
Search using starting postion & searching length: -1
Search using starting postion & string comparision: 15
```

## Codebyte Example

> In this example, the `String.LastIndexOf()` is use with `compareStr`, `startIndex` and  `strCmp` parameters to determine the zero-base index of the last occurrence of substring within the string. The search starts at the startIndex and proceeds backward toward the beginning of the string, using the StringComparison method:

Reviewer: ```suggestion
In this example, the `String.LastIndexOf()` is used with `compareStr`, `startIndex` and  `strCmp` parameters to determine the zero-base index of the last occurrence of substring within the string. The search starts at the startIndex and proceeds backward toward the beginning of the string, using the StringComparison method:
```

```codebyte/csharp
using System;

public class HelloWorld
{
    public static void Main(string[] args)
    {
        string baseString = "Hello, I'm Codecademy";

>         compareStr = "C";

Reviewer 2: ```suggestion
        string compareStr = "C";
```
